src/plugins/workspace/public/components/workspace_creator/utils.ts

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { WorkspacePermissionMode } from '../../../../../core/public';
+import {
+  UserOrGroupPermissionEditingData,
+  WorkspaceFormEditingData,
+  WorkspaceFormErrors,
+  WorkspacePermissionSetting,
+  WorkspacePermissionItemType,
+} from './types';
+import { OptionIdToWorkspacePermissionModesMap, PermissionModeId } from './constants';
+
+export const isValidWorkspacePermissionSetting = (
+  setting: Partial<WorkspacePermissionSetting>
+): setting is WorkspacePermissionSetting =>
+  !!setting.modes &&
+  setting.modes.length > 0 &&
+  ((setting.type === WorkspacePermissionItemType.User && !!setting.userId) ||
+    (setting.type === WorkspacePermissionItemType.Group && !!setting.group));
+
+export const getErrorsCount = (formErrors: WorkspaceFormErrors) => {
+  let errorsCount = 0;
+  if (formErrors.name) {
+    errorsCount += 1;
+  }
+  if (formErrors.description) {
+    errorsCount += 1;
+  }
+  if (formErrors.permissions) {
+    errorsCount += formErrors.permissions.length;
```

Comment:
As I remember the formErrors.permissions was initialized with `new Array()`, the length will always be permissions length.

Reply:
Thanks for point out this bug. I will fix this.